zetaclient/chains/ton/config.go

```diff
@@ -52,3 +64,46 @@ func ConfigFromSource(ctx context.Context, urlOrPath string) (*GlobalConfigurati
 
 	return ConfigFromPath(urlOrPath)
 }
+
+// FetchGasConfig fetches gas price from the config.
+func FetchGasConfig(ctx context.Context, getter ConfigGetter) (tlb.GasLimitsPrices, error) {
+	// https://docs.ton.org/develop/howto/blockchain-configs
+	// https://tonviewer.com/config#21
+	const configKeyGas = 21
+
+	response, err := getter.GetConfigParams(ctx, 0, []uint32{configKeyGas})
+	if err != nil {
+		return tlb.GasLimitsPrices{}, errors.Wrap(err, "failed to get config params")
+	}
+
+	ref, ok := response.Config.Get(configKeyGas)
+	if !ok {
+		return tlb.GasLimitsPrices{}, errors.Errorf("config key %d not found", configKeyGas)
+	}
+
+	var cfg tlb.ConfigParam21
+	if err = tlb.Unmarshal(&ref.Value, &cfg); err != nil {
+		return tlb.GasLimitsPrices{}, errors.Wrap(err, "failed to unmarshal config param")
+	}
+
+	return cfg.GasLimitsPrices, nil
+}
+
+// ParseGasPrice parses gas price from the config and returns price in tons per 1 gas unit.
+func ParseGasPrice(cfg tlb.GasLimitsPrices) (uint64, error) {
+	// from TON docs: gas_price: This parameter reflects
+	// the price of gas in the network, in nano tons per 65536 gas units (2^16).
+	switch cfg.SumType {
+	case "GasPrices":
```

Comment:
Can't we use enum values for these?

Reply:
technically we can, but `tongo` package doesn't use them. I'll move this to a `const`